modules/oceanbase-ce/src/main/java/org/testcontainers/containers/OceanBaseContainer.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    @Override
+    public String getDriverClassName() {
+        return driverClassName;
+    }
+
+    @Override
+    public String getJdbcUrl() {
+        return getJdbcUrl(DEFAULT_DATABASE_NAME);
+    }
+
+    public String getJdbcUrl(String databaseName) {
+        String additionalUrlParams = constructUrlParameters("?", "&");
+        String prefix = driverClassName.contains("mysql") ? "jdbc:mysql://" : "jdbc:oceanbase://";
+        return prefix + getHost() + ":" + getMappedPort(SQL_PORT) + "/" + databaseName + additionalUrlParams;
+    }
+
+    @Override
+    public String getDatabaseName() {
+        return DEFAULT_DATABASE_NAME;
+    }
+
+    @Override
+    public String getUsername() {
+        return DEFAULT_USERNAME + "@" + tenantName;
+    }
+
+    @Override
+    public String getPassword() {
+        return DEFAULT_PASSWORD;
+    }
+
+    @Override
+    protected String getTestQueryString() {
+        return "SELECT 1";
+    }
+
+    /**
+     * Enable fastboot.
+     *
+     * @return this
+     */
+    public OceanBaseContainer enableFastboot() {
+        this.enableFastboot = true;
+        return self();
+    }
+
+    /**
+     * Set the deployment mode, see <a href="https://hub.docker.com/r/oceanbase/oceanbase-ce">Docker Hub</a> for more details.
+     *
+     * @param mode the deployment mode
+     * @return this
+     */
+    public OceanBaseContainer withMode(String mode) {
+        this.mode = mode;
+        return self();
+    }
+
+    /**
+     * Set the non-system tenant to be created for testing.
+     *
+     * @param tenantName the name of tenant to be created
+     * @return this
+     */
+    public OceanBaseContainer withTenant(String tenantName) {
+        if (StringUtils.isEmpty(tenantName)) {
+            throw new IllegalArgumentException("Tenant name cannot be null or empty");
+        }
+        if (SYSTEM_TENANT_NAME.equals(tenantName)) {
+            throw new IllegalArgumentException("Tenant name cannot be " + SYSTEM_TENANT_NAME);
+        }
+        this.tenantName = tenantName;
+        return self();
+    }
+
+    /**
+     * Set the driver class name.
+     *
+     * @param driverClassName the driver class name
+     * @return this
+     */
+    public OceanBaseContainer withDriverClassName(String driverClassName) {
+        if (StringUtils.isEmpty(driverClassName)) {
+            throw new IllegalArgumentException("Driver class name cannot be null or empty");
+        }
+        if (!driverClassName.contains("mysql") && !driverClassName.contains("oceanbase")) {
+            throw new IllegalArgumentException("Driver class name should contains 'mysql' or 'oceanbase'");
+        }
+        try {
+            Class.forName(driverClassName);
+        } catch (ClassNotFoundException e) {
+            throw new IllegalArgumentException("Driver class not found", e);
+        }
+        this.driverClassName = driverClassName;
+        return self();
+    }
```

Comment:
Usually, driver are detected as in other implementations because those are related to the Testcontainers JDBC URL support.